{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/base/api/mixins.py

```diff
@@ -29,3 +29,16 @@ def my_action:
         if self.action in self.serializer_classes.keys():
             return self.serializer_classes[self.action]
         return super().get_serializer_class()
+
+
+class PermissionPerActionMixin:
+    attribute = "permissions_per_action"
+
+    def get_permissions(self):
+        if not isinstance(getattr(self, self.attribute), dict):
+            raise ImproperlyConfigured(f"{self.attribute} should be a dict mapping.")
+
+        if self.action in getattr(self, self.attribute).keys():
+            self.permission_classes = getattr(self, self.attribute)[self.action]
```

Comment:
```suggestion
            return action_permissions[self.action]
```

Reply:
permission classes have a different api than serializer classes. Hence, we would need to modify the `self.permission_classes` instead of returning them.

```
    def get_permissions(self):
        """
        Instantiates and returns the list of permissions that this view requires.
        """
        return [permission() for permission in self.permission_classes]

```

Reply:
cool.